### PR TITLE
Add memory leak integration test for Exec Sync

### PR DESCRIPTION
**Pull request by saschagrunert**
#### What type of PR is this?

/kind bug

#### What this PR does / why we need it:

The test unveils a memory leak in Exec Sync without providing a fix yet.

#### Which issue(s) this PR fixes:

None
#### Special notes for your reviewer:
None
#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

File: pkg/client/client_test.go [hunk 1]
```diff
@@ -247,6 +248,37 @@ var _ = Describe("ConmonClient", func() {
 				}
 				wg.Wait()
 			})
+
+			It(testName("should not leak memory", terminal), func() {
+				tr = newTestRunner()
+				tr.createRuntimeConfigWithProcessArgs(terminal, []string{"/busybox", "sleep", "60"}, nil)
```
[the remark this answers is not in the document]
Comment (saschagrunert) [reply pkg/client/client_test.go]: Yep sure, I can bump it to 600.

File: pkg/client/client_test.go [hunk 2]
```diff
@@ -247,6 +248,37 @@ var _ = Describe("ConmonClient", func() {
...
+						ID:       tr.ctrID,
+						Command:  []string{"/busybox", "echo", "-n", "hello", "world", fmt.Sprintf("%d", i)},
+						Terminal: terminal,
+						Timeout:  timeoutUnlimited,
+					})
+
+					Expect(err).To(BeNil())
+					Expect(result).NotTo(BeNil())
+					Expect(string(result.Stdout)).To(Equal(fmt.Sprintf("hello world %d", i)))
+					fmt.Println("done with", i, string(result.Stdout))
```
Comment (haircommander) [inline pkg/client/client_test.go:274]: need to update to GinkgoWriter here

Comment (saschagrunert) [reply pkg/client/client_test.go]: Yep, fixed :+1:

**Comment by saschagrunert** [comment]
The `read_loop_stdin` of `container_io.rs` caused the problem. I rebased on top of #683 which fixes the problem magically.

**Pull request merged**